apps/alpha-test-app/src/main.ts

```diff
@@ -5,11 +5,15 @@ import App from "./App.vue";
 import { createI18n } from "vue-i18n";
 import enUS from "./i18n/locales/en-US.json";
 import deDE from "./i18n/locales/de-DE.json";
+import onyxDeDE from "sit-onyx/locales/de-DE.json";
+import { createOnyx } from "sit-onyx";
 
 const i18n = createI18n({
   legacy: false,
   locale: "en-US",
   messages: { "en-US": enUS, "de-DE": deDE },
 });
 
-createApp(App).use(i18n).mount("#app");
+const onyx = createOnyx({ i18n: { locale: i18n.global.locale, messages: { "de-DE": onyxDeDE } } });
```

Comment:
I see two issues with the plugin approach: Its harder/impossible to use this feature with nuxt or when building your vue app as custom element

Imagine this use case:
- you have an existing website that was created with a different tech stack (maybe not even vue)
- you now want to insert one custom element (e.g. a complex form) into this website so you need to build your app as custom element so styles are scoped and not effected by the existing website
- there is no app instance in this case so you can not use Vue plugins

Imho we should support those use cases. Maybe we can just switch from a Vue plugin to a regular function that is called in the App.vue like it was before? Something like `provideOnyx()`

Reply:
I wouldn't focus to much on microfrontend support, but Nuxt support is a good point!
This is *out-of-scope* for now, but I created a follow-up ticket: [#328](https://github.com/SchwarzIT/onyx/issues/328)